src/core/components/pubsub.js

```diff
@@ -20,13 +19,19 @@ module.exports = function pubsub (self) {
             resolve()
           })
         })
-      } else {
-        self._libp2pNode.pubsub.subscribe(topic, options, handler, callback)
       }
+
+      self._libp2pNode.pubsub.subscribe(topic, options, handler, callback)
     },
 
-    unsubscribe: (topic, handler) => {
+    unsubscribe: (topic, handler, callback) => {
       self._libp2pNode.pubsub.unsubscribe(topic, handler)
+
+      if (!callback) {
+        return Promise.resolve()
+      }
+
+      process.nextTick(() => callback())
```

Comment:
not all bundlers know how to handle process.nextTick well. We always use `async/setImmediate` for that matter.